client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -234,6 +235,17 @@ static Request flush(FlushRequest flushRequest) {
         return new Request(HttpPost.METHOD_NAME, endpoint, parameters.getParams(), null);
     }
 
+    static Request clearCache(ClearIndicesCacheRequest clearIndicesCacheRequest) {
+        String endpoint = endpoint(clearIndicesCacheRequest.indices(), "_cache", "clear");
+        Params parameters = Params.builder();
+        parameters.withIndicesOptions(clearIndicesCacheRequest.indicesOptions());
+        parameters.putParam("query", Boolean.toString(clearIndicesCacheRequest.queryCache()));
+        parameters.putParam("fielddata", Boolean.toString(clearIndicesCacheRequest.fieldDataCache()));
```

Comment:
I think the `fielddata` thing is a mistake in the way we use `ParseField` to read parameters. Looking at the history, they are just synonyms, we never deprecated one in favour of the other. We use `fielddata` in our docs though. You are welcome to send a PR to remove support for the other deprecated params against master though ;)

Reply:
Sorry, I only went with the code ( without going into details ).

However, I just called `GET _cache/clear?fielddata` in Kibana and got 
```
#! Deprecation: Deprecated field [fielddata] used, expected [field_data] instead
{
  "_shards": {
    "total": 2,
    "successful": 1,
    "failed": 0
  }
}
```
The deprecation seems to be wrapped in the [ParseField#match](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/common/ParseField.java#L105)

>We use `fielddata` in our docs though. 

Good catch! Maybe it would be better to have a separate PR for the ref docs ( so that we can track when the preferred name has been changed and since when a deprecation msg is logged ) ( I can take this one over if you want )

> You are welcome to send a PR to remove support for the other deprecated params against master though ;)

Which other params do you mean? And remove them from the the rest api specs and the ref docs ?

Reply:
> Which other params do you mean? And remove them from the the rest api specs and the ref docs ?

I meant `filter`, `filter_cache`, and `request_cache`. I think we can remove them now in master.

Reply:
`fielddata` is the preferred name as of my merging #28943 today.